src/utils.ts

```diff
@@ -6,6 +6,52 @@
  */
 import { ContextAttributes } from './constructor-types';
 
+function chceksum(obj: any): string {
+  const str = JSON.stringify(obj);
+  let hash = 0;
+  for (let i = 0; i < str.length; i++) {
+    const chr = str.charCodeAt(i);
+    hash = (hash << 5) - hash + chr;
+    hash |= 0;
+  }
+  return hash.toString(16);
+}
+
+/**
+ * a class that hold an array of objects that each object would be unique
+ *
+ */
+class ObjectSet {
+  dataArray: any[] = [];
+  dataDict: any = {};
+
+  addMany(objects: any[]) {
+    objects.forEach((obj) => this.add(obj));
+  }
+
+  clear() {
+    this.dataArray = [];
+    this.dataDict = {};
+  }
+
+  add(obj: any) {
```

Comment:
obj: Record<string, unknown> | unknown
probably, if I understood. the logic correctly